components/director/cmd/tenantfetcher-job/main.go

```diff
@@ -123,7 +126,10 @@ func createTenantFetcherSvc(cfg config, transact persistence.Transactioner, kube
 
 	runtimeService := runtime.NewService(runtimeRepository, labelRepository, labelDefService, labelService, uidSvc, scenarioAssignEngine, tenantStorageSvc, cfg.Features.ProtectedLabelPattern, cfg.Features.ImmutableLabelPattern)
 
-	eventAPIClient := tenantfetcher.NewClient(cfg.OAuthConfig, cfg.APIConfig, cfg.ClientTimeout)
+	eventAPIClient, err := tenantfetcher.NewClient(cfg.OAuthConfig, cfg.AuthMode, cfg.APIConfig, cfg.ClientTimeout)
+	if nil != err {
+		exitOnError(err, "failed to create event-api client")
+	}
```

Comment:
No need of the if can directly invoke exitOnError has the same check inside.

Reply:
The There are several exit points inside `NewClient`. On the other hand the rest of the methods return an `error` and do `exitOnError()` in main. I'll propagate the error just to be consistent with the rest of the project.